### ISSUE TITLE
Support custom logic of `getNextVersion`

### ISSUE THREAD

**Issue by azu** [algolia/shipjs#713]
**Is your feature request related to a problem? Please describe.**

Currently, It seems that ship.js support only conventional commits to get Next version.

- https://github.com/algolia/shipjs/blob/5c5cca180a37a163bef53df28d222c923eab87c2/packages/shipjs-lib/src/lib/util/getNextVersion.js

Can I custom the detection logic for getting next version?

I want to custom the logic of `getNextVersion`.
For example, I want to implement label based version management:

> Calculate Next version from GitHub PR's labels
>    If these includes only `Version: patch`, next version is patch
>    If these includes `Version: patch` and `Version: minor`,  next version is minor
>    If any one is `Version: major` label, next version is major
>   --  https://gist.github.com/azu/e39fcc506eff4242beb27b84cac17b82

During I try to implement [my Release Flow idea](https://gist.github.com/azu/e39fcc506eff4242beb27b84cac17b82) using ship.js, I noticed that ship.js only support conventional commits to detect next version.

**Describe the solution you'd like**

Add new lifecyle hook like `updateVersion` before ` shouldPrepare`

```js
module.exports = {
    updateVersion: async ({
                     revisionRange,
                     currentVersion,
                     dir}) => {
       // user defined logic
       const nextVersion = await calculateNextVersion();
       return { nextVersion } 
    }
};
```

- https://community.algolia.com/shipjs/reference/all-config.html

**Describe alternatives you've considered**

- Options 2: Add `{ versionStrategy: "conventional-commits" | customFunction }`
- Options 3: Load next version from file if exist
  - Example: If exists `.next-version` file in the repository, use it as next version and delete it after prepared
  - This approach is similar with [changesets](https://github.com/atlassian/changesets)

I don't know ship.js internal, maybe exist more reasonable implementation.

**Additional context**

**Comment by eunjae-lee** [comment]
Hi @azu ,
I guess we can expose a config so that you can override `getNextVersion` function.

I will try to work on this this week or next week.
Or, if you want to contribute on the code level, let me know. I will help you onboard!

**Comment by eunjae-lee** [comment]
Hi @azu, this has been released at v0.18.0.
[Doc ->](https://community.algolia.com/shipjs/reference/all-config.html#getnextversion)

**Comment by azu** [comment]
Thanks!
